src/sirutils.c

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file sirutils.c
```

Comment:
1. utils is a _very_ generic term to use for something specific to string manipulation. I would rename it `sirstrmanip` or something more explicit.
2. if you want these to be part of the public interface (i.e., don't begin with underscore), then they have to be fully Doxygen'd and the header included by `sir.h`. if you are unsure of your abilities, you can write synopsis for me in a text file or something and I can do the doxygen comments.

Reply:
I'll just move to helpers as discussed.